azure-communication-ui/azure-communication-ui/src/calling/java/com/azure/android/communication/ui/calling/presentation/manager/AudioSessionManager.kt

```diff
@@ -112,26 +112,37 @@ internal class AudioSessionManager(
         if (activity.isFinishing) {
             btAdapter?.run {
                 closeProfileProxy(BluetoothProfile.HEADSET, bluetoothAudioProxy)
+                isProfileProxyOpened = false
             }
-            context.unregisterReceiver(this)
 
             if (audioManager.isBluetoothScoOn) {
                 audioManager.stopBluetoothSco()
             }
             audioManager.isBluetoothScoOn = false
             audioManager.isSpeakerphoneOn = false
+
+            context.unregisterReceiver(this@AudioSessionManager)
         }
     }
 
     override fun onReceive(context: Context?, intent: Intent?) {
         intent?.apply {
             when (action) {
-                BluetoothHeadset.ACTION_CONNECTION_STATE_CHANGED -> updateBluetoothStatus()
+                BluetoothAdapter.ACTION_CONNECTION_STATE_CHANGED -> openProfileProxy()
                 AudioManager.ACTION_HEADSET_PLUG -> updateHeadphoneStatus()
             }
         }
     }
 
+    private fun openProfileProxy() {
+        if (!isProfileProxyOpened && btAdapter?.state == BluetoothAdapter.STATE_ON) {
+            btAdapter?.run {
+                getProfileProxy(context, this@AudioSessionManager, BluetoothProfile.HEADSET)
```

Comment:
what should happen if `getProfileProxy()` returns false?

Reply:
ok, I looked at the SDK source code for `getProfileProxy()`, ignore my above comment, it should actually be returning a reference to the BT Headset object, but it doesn't.